Build/libHttpClient.Android/src/main/java/com/xbox/httpclient/HttpClientRequestBody.java

```diff
@@ -48,7 +48,8 @@ public int read(byte[] destination, int destinationOffset, int length) throws IO
             }
 
             int bytesRead = nativeRead(this.callHandle, this.offset, destination, destinationOffset, length);
-            if (bytesRead == 0) {
```

Comment:
should it be 0 or -1 ?

Reply:
if we return 0 bytes read that probably represents a bug, and i lean towards making this listen explicitly for -1 so that if there is that bug we get into trouble around it rather than a potentially more opaque kind of trouble by telling the OS prematurely that we're done reading. Open to changing it, though.

Reply:
alright, that's fine